### PR TITLE
chore: Secretrefs refactoring

### PR DESCRIPTION
**Pull request by TeodorSAP**
## Description

Changes proposed in this pull request (what was done and why):

- Move secretrefs functionality out of the API definitions and into the secretref package

Changes refer to particular issues, PRs or documents:

- Prerequisite for: https://github.com/kyma-project/telemetry-manager/issues/872

## Traceability
- [x] The PR is linked to a GitHub issue.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [ ] If the change is user-facing, the documentation has been adjusted.
- [ ] If a CRD is changed, the corresponding Busola ConfigMap has been adjusted.
- [x] The feature is unit-tested.
- [x] The feature is e2e-tested.

### CODE REVIEW

File: internal/validators/secretref/secret_ref.go [hunk 1]
```diff
@@ -84,3 +91,101 @@ func checkForMissingFields(ref telemetryv1alpha1.SecretKeyRef) error {
...
+	}
+
+	refs = append(refs, GetEnvSecretRefs(lp)...)
+	refs = append(refs, getTLSSecretRefs(lp)...)
+
+	return refs
+}
+
+// GetEnvSecretRefs returns the secret references of a LogPipeline that should be stored in the env secret
+func GetEnvSecretRefs(lp *telemetryv1alpha1.LogPipeline) []telemetryv1alpha1.SecretKeyRef {
```
Comment (skhalash) [inline internal/validators/secretref/secret_ref.go:119]: Does it have to be exported?

File: internal/validators/secretref/secret_ref.go [hunk 2]
```diff
@@ -29,8 +25,19 @@ var (
...
+}
+
+func (v *Validator) ValidateMetricPipeline(ctx context.Context, pipeline *telemetryv1alpha1.MetricPipeline) error {
+	return v.validate(ctx, getSecretRefsMetricPipeline(pipeline))
+}
+
+func (v *Validator) ValidateLogPipeline(ctx context.Context, pipeline *telemetryv1alpha1.LogPipeline) error {
+	return v.validate(ctx, getSecretRefsLogPipeline(pipeline))
+}
+
```
Comment (skhalash) [inline internal/validators/secretref/secret_ref.go:39]: Please add godoc comments

**Review by NHingerl — DISMISSED** [review]

File: internal/validators/secretref/secret_ref.go (hunk 1 above)
Comment (TeodorSAP) [reply internal/validators/secretref/secret_ref.go]: I've re-implemented the logic in `internal/reconciler/logpipeline/fluentbit/sync.go`, avoiding the export of this function.

File: internal/validators/secretref/secret_ref.go [hunk 3]
```diff
@@ -84,3 +97,101 @@ func checkForMissingFields(ref telemetryv1alpha1.SecretKeyRef) error {
...
+	if output.HTTP != nil {
+		refs = appendIfSecretRef(refs, output.HTTP.Host)
+		refs = appendIfSecretRef(refs, output.HTTP.User)
+		refs = appendIfSecretRef(refs, output.HTTP.Password)
+	}
+
+	return refs
+}
+
+func getTLSSecretRefs(lp *telemetryv1alpha1.LogPipeline) []telemetryv1alpha1.SecretKeyRef {
```
Comment (skhalash) [inline internal/validators/secretref/secret_ref.go:138]: You already have a function `getSecretRefsInOTLPOutput`. Maybe just merge those 2 functions into `getSecretRefsInHTTPOutput` for the sake of consistency?

**Review by skhalash — DISMISSED** [review]

**Review by skhalash — APPROVED** [review]

**Review by NHingerl — APPROVED** [review]

**Comment by NHingerl** [comment]
did not update DITA because the only change in the .md seems to have been table column adjustment

**Pull request merged**